[iOS] Form accessory view buttons may overflow the toolbar

pxlcoder opened the pull request:
#### bab2730bdb34a67ef10805b767893183e8aab941
<pre>
[iOS] Form accessory view buttons may overflow the toolbar
<a href="https://bugs.webkit.org/show_bug.cgi?id=291620">https://bugs.webkit.org/show_bug.cgi?id=291620</a>
<a href="https://rdar.apple.com/147270681">rdar://147270681</a>

Reviewed by Abrar Rahman Protyasha.

Add a margin to the toolbar when necessary. Additionally, reduce spacing between
the next and previous buttons if a margin is present.

* Source/WebKit/UIProcess/ios/forms/WKFormAccessoryView.mm:
(-[WKFormAccessoryView _toolbarMargin]):
(-[WKFormAccessoryView initWithInputAssistantItem:delegate:]):
(-[WKFormAccessoryView _updateFrame]):
(-[WKFormAccessoryView setNextPreviousItemsVisible:]):

Update logic to account for the lack of a spacer.

Canonical link: <a href="https://commits.webkit.org/293778@main">https://commits.webkit.org/293778@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f58b4cc1faa970438cb1276c9b38f4f5d913835c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/99863 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/19511 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/9794 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/104990 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/50444 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/19816 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/27946 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/76023 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/33113 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/102870 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/15112 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/90184 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/56381 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14917 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/49813 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/84844 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/8258 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/107351 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/26976 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/19714 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/84982 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/27338 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/86384 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/84504 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/29174 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/6899 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/20794 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16252 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/26913 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/32128 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/26724 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/30040 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/28283 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/UIProcess/ios/forms/WKFormAccessoryView.mm:
```diff
@@ -175,6 +188,24 @@ - (instancetype)initWithInputAssistantItem:(UITextInputAssistantItem *)inputAssi
+            [[_rightToolbar leadingAnchor] constraintEqualToAnchor:[[_rightToolbar superview] leadingAnchor]],
+            [[_rightToolbar trailingAnchor] constraintEqualToAnchor:[[_rightToolbar superview] trailingAnchor]],
+            [[_rightToolbar topAnchor] constraintEqualToAnchor:[[_rightToolbar superview] topAnchor] constant:toolbarMargin],
+            [[_rightToolbar bottomAnchor] constraintEqualToAnchor:[[_rightToolbar superview] bottomAnchor] constant:-toolbarMargin],
```
Nit: Scope to write a lambda and call it with _left/rightToolbar here

aprotyas replied:
Not that it's saving you a whole lot of repetition

pxlcoder replied:
Done.

pxlcoder committed:
[iOS] Form accessory view buttons may overflow the toolbar
https://bugs.webkit.org/show_bug.cgi?id=291620
rdar://147270681

Reviewed by Abrar Rahman Protyasha.

Add a margin to the toolbar when necessary. Additionally, reduce spacing between
the next and previous buttons if a margin is present.

* Source/WebKit/UIProcess/ios/forms/WKFormAccessoryView.mm:
(-[WKFormAccessoryView _toolbarMargin]):
(-[WKFormAccessoryView initWithInputAssistantItem:delegate:]):
(-[WKFormAccessoryView _updateFrame]):
(-[WKFormAccessoryView setNextPreviousItemsVisible:]):

Update logic to account for the lack of a spacer.

Canonical link: https://commits.webkit.org/293778@main
